echopype/tests/test_preprocess.py

```diff
@@ -0,0 +1,151 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+import echopype as ep
+
+
+def test_remove_noise():
+    """Test remove_noise on toy data"""
+
+    # Parameters for fake data
+    nfreq, npings, nrange = 1, 10, 100
+    freq = np.arange(nfreq)
+    ping_index = np.arange(npings)
+    range_bin = np.arange(nrange)
+    data = np.ones(nrange)
+
+    # Insert noise points
+    np.put(data, 30, -30)
+    np.put(data, 60, -30)
+    # Add more pings
+    data = np.array([data] * npings)
+    # Make DataArray
+    Sv = xr.DataArray([data], coords=[('frequency', freq),
+                                      ('ping_time', ping_index),
+                                      ('range_bin', range_bin)])
+    Sv.name = "Sv"
+    ds_Sv = Sv.to_dataset()
+
+    ds_Sv = ds_Sv.assign(range=xr.DataArray(np.array([[np.linspace(0, 10, nrange)] * npings]), coords=Sv.coords))
+    ds_Sv = ds_Sv.assign(sound_absorption=0.001)
+    # Run noise removal
+    ds_Sv = ep.preprocess.remove_noise(ds_Sv, ping_num=2, range_bin_num=5, SNR_threshold=0)
+
+    # Test if noise points are are nan
+    assert np.isnan(ds_Sv.Sv_corrected.isel(frequency=0, ping_time=0, range_bin=30))
+    assert np.isnan(ds_Sv.Sv_corrected.isel(frequency=0, ping_time=0, range_bin=60))
+
+    # Test remove noise on a normal distribution
+    np.random.seed(1)
+    data = np.random.normal(loc=-100, scale=2, size=(nfreq, npings, nrange))
+    # Make Dataset to pass into remove_noise
+    Sv = xr.DataArray(data, coords=[('frequency', freq),
+                                    ('ping_time', ping_index),
+                                    ('range_bin', range_bin)])
+    Sv.name = "Sv"
+    ds_Sv = Sv.to_dataset()
+    # Attach required range and sound_absorption values
+    ds_Sv = ds_Sv.assign(range=xr.DataArray(np.array([[np.linspace(0, 3, nrange)] * npings]), coords=Sv.coords))
+    ds_Sv = ds_Sv.assign(sound_absorption=0.001)
+    # Run noise removal
+    ds_Sv = ep.preprocess.remove_noise(ds_Sv, ping_num=2, range_bin_num=5, SNR_threshold=0)
+    null = ds_Sv.Sv_corrected.isnull()
+    # Test to see if the right number of points are removed before the range gets too large
+    assert np.count_nonzero(null.isel(frequency=0, range_bin=slice(None, 50))) == 6
+
+
+def test_compute_MVBS_index_binning():
+    """Test compute_MVBS_index_binning on toy data"""
+
+    # Parameters for fake data
+    nfreq, npings, nrange = 4, 40, 400
+    ping_num = 2             # number of pings to average over
+    range_bin_num = 3        # number of range_bins to average over
```

Comment:
Similar to the comment on `test_compute_MVBS` below, could you change `ping_num` or `nrange` so that `nrange` is not an integer multiple of `ping_num`? 

This will help us test that the `xr.coarsen(..., boundary='pad').mean()` was done correctly. It likely is already since along the `range_bin` dimension the reduction is not an integer multiple.